gc.c

```diff
@@ -15187,6 +15218,11 @@ void rb_mmtk_post_process_opts(char *s) {
     if (opt_match_arg(s, l, "plan")) {
         mmtk_post_arg_plan = s + 1;
     }
+    if (opt_match_arg(s, l, "max-heap")) {
+        // this has already been processed in order to configure the heap.
+        // we need it here to define it as a valid arg
```

Comment:
The other values - like the plan, get verified (the next function I think) so that if we break argument pre-processing we'll find out.

Reply:
I've added a seperate boolean to track whether a custom heap size has been passed, so that we can use that boolean to verify `mmtk_max_heap_size` in `rb_mmtk_post_process_opts_finish`. Is this the sort of thing you had in mind?